### DISCUSSION TITLE
Repeated Indexing for europe continent database

### DISCUSSION THREAD

**Discussion by akki401** [osm-search/Nominatim#3429] [General]
Hello I see the database indexing is repeated.
After RANK 30 ETA I see again it start indexing the from RANK 0
Any problem ?
![Screenshot 2024-05-25 084317](https://github.com/osm-search/Nominatim/assets/50397166/66e8123a-a8a9-42a5-b836-a4a32102a9ee)

**EC2 config:**
i4i.2xlarge
vCPU = 8
Memory (GiB) = 64
Instance Storage (GB) = 1 x 1,875 AWS Nitro SSD
Network Bandwidth (Gbps) = Up to 12
EBS Bandwidth (Gbps) = Up to 10

**Docker command:**
cmd = " ".join(
[
"docker",
"run",
"--restart=always",
"--log-driver json-file",
"--log-opt max-size=10m",
"--log-opt max-file=3",
"--shm-size=4g",
"-e",
"POSTGRES_SHARED_BUFFERS=2GB",
"-e",
"POSTGRES_MAINTAINENCE_WORK_MEM=10GB",
"-e",
"POSTGRES_AUTOVACUUM_WORK_MEM=2GB",
"-e",
"POSTGRES_WORK_MEM=50MB",
"-e",
"POSTGRES_EFFECTIVE_CACHE_SIZE=24GB",
"-e",
"POSTGRES_SYNCHRONOUS_COMMIT=off",
"-e",
"POSTGRES_MAX_WAL_SIZE=1GB",
"-e",
"POSTGRES_CHECKPOINT_TIMEOUT=10min",
"-e",
"POSTGRES_CHECKPOINT_COMPLETITION_TARGET=0.9",
"-e",
f"PBF_PATH=/nominatim/data/{self.region}-latest.osm.pbf",
"-e",
f"REPLICATION_URL={self._updates_url}", # source url of the updates for this area
"-e",
f"REPLICATION_UPDATE_INTERVAL={3600 * 24 * 30}", # How often upstream publishes diffs

"-e",
f"REPLICATION_RECHECK_INTERVAL={3600 * 24 * 15}", # How long to sleep if no update found yet

"-e",
"UPDATE_MODE=none", # continuous/once/catch-up/none
"-e",
"FREEZE=false",
"-e",
"IMPORT_STYLE=full",
"-e",
"NOMINATIM_FLATNODE_FILE=/nominatim/flatnode/flatnode.file",
"-v",
f"{self._dir_nominatim_postgressql_db}/:/var/lib/postgresql/14/main",
"-v",
f"{self._dir_nominatim}/:/nominatim/data",
"-v",
f"{self._dir_nominatim_flatnode}/:/nominatim/flatnode",
"-p",
f"{self._nominatim_port}:8080",
"-i",
"-t",
"-d",
"--name",
self._nominatim_container_name,
"mediagis/nominatim:4.4",
]
)

**Comment by lonvia** [comment]
It's not repeated. The order of execution just happens to be so that rank 0 comes last after ranks 1 to 30.
